packages/tools/src/tools/annotation/PlanarFreehandROITool.ts

```diff
@@ -0,0 +1,446 @@
+import {
+  getEnabledElement,
+  triggerEvent,
+  eventTarget,
+  Settings,
+} from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { Events } from '../../enums';
+import { AnnotationTool } from '../base';
+import {
+  addAnnotation,
+  getAnnotations,
+} from '../../stateManagement/annotation/annotationState';
+import { polyline } from '../../utilities/math';
+import { getViewportIdsWithToolToRender } from '../../utilities/viewportFilters';
+import triggerAnnotationRenderForViewportIds from '../../utilities/triggerAnnotationRenderForViewportIds';
+import registerDrawLoop from './planarFreehandROITool/drawLoop';
+import registerEditLoopCommon from './planarFreehandROITool/editLoopCommon';
+import registerClosedContourEditLoop from './planarFreehandROITool/closedContourEditLoop';
+import registerOpenContourEditLoop from './planarFreehandROITool/openContourEditLoop';
+import registerOpenContourEndEditLoop from './planarFreehandROITool/openContourEndEditLoop';
+import registerRenderMethods from './planarFreehandROITool/renderMethods';
+import {
+  AnnotationCompletedEventDetail,
+  AnnotationModifiedEventDetail,
+} from '../../types/EventTypes';
+import {
+  EventTypes,
+  ToolHandle,
+  PublicToolProps,
+  ToolProps,
+  InteractionTypes,
+} from '../../types';
+import { PlanarFreehandROIAnnotation } from '../../types/ToolSpecificAnnotationTypes';
+import { PlanarFreehandROICommonData } from '../../utilities/math/polyline/planarFreehandROIInternalTypes';
+
+const { pointCanProjectOnLine } = polyline;
+
+/**
+ * PlanarFreehandROITool lets you draw annotations that define an arbitrarily drawn region.
+ * You can use the PlanarFreehandROITool in all perpendicular views (axial, sagittal, coronal),
```

Comment:
Is the comment correct? Does it support all perpendicular views?

Reply:
Yeah, the only reason it doesn't support obliques actually is just because of the implementation in `getSpacingAndXYDirections`, which could late be improved.

When I say all perpendicular views, I mean the same as all other planar annotations. These are just planar contours for now, so no rendering of planar annotations drawn in one view, in another view, same as e.g. Length.

Reply:
I'll clarify the comment.